flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/FlinkSource.java

```diff
@@ -151,9 +157,19 @@ public DataStream<RowData> build() {
           scanContext.caseSensitive(),
           arcticTable.io()
       );
+
+      RowType rowType;
+      if (projectedSchema != null) {
+        rowType = toRowType(projectedSchema);
+      } else {
+        rowType = FlinkSchemaUtil.convert(scanContext.project());
+      }
+
+      boolean generateWatermark = PropertyUtil.propertyAsBoolean(properties, ARCTIC_WATERMARK.key(),
+          ARCTIC_WATERMARK.defaultValue());
       return env.fromSource(
```

Comment:
I think we should add a new ConfigOption to control the generation of the watermark strategy. Otherwise, all watermark configured arctic table would use the custom watermark strategy.